packages/eslint-plugin/src/configs/core.js

```diff
@@ -19,7 +19,7 @@ const config = {
       // @see https://github.com/alexgorbatchev/eslint-import-resolver-typescript#configuration
       typescript: {
         alwaysTryTypes: true,
-        project: './tsconfig.json',
```

Comment:
this will increase lint speed by huge amount. behind the scenes if there is ts solution config used it will be unwrapped by this resolver and all referenced tsconfigs will be loaded. I'd suggest to investigate vscode eslint configuration. from CLI perspective everything works as expected.

Reply:
Thanks for the reply. This is out of my area of expertise, so forgive any misunderstandings on my part. I was going to log a bug about the errors in the dev environment, but I first did some investigation and figured I'd submit a PR with a potential fix instead. Here's some more details about why I figured this was the right approach.

I noticed your RFC about the project config mentioned that all of the additional project configs need to be specified in the `"project"` field since eslint doesn't support project references:
https://github.com/microsoft/fluentui/blob/master/rfcs/shared/build-system/02-solution-style-ts-configs.md#eslint-setup
```
    // we need to specify our environment configs (will be removed once typescript-eslint properly supports project references)
    // tsc program will be created accordingly per environment/scope
    "project": [
      // includes tsconfig.lib.json and tsconfig.spec.json
      "path/to/<package-name>/tsconfig.*?.json",
      // includes storybook config
      "path/to/<package-name>/.storybook/tsconfig.json"
    ]
```

The typescript-eslint docs also recommend adding all projects from the tsconfig's references to the eslint's project: https://typescript-eslint.io/architecture/parser/#project
> If you use project references, TypeScript will not automatically use project references to resolve files. This means that you will have to add each referenced tsconfig to the project field either separately, or via a glob.

--

It looks like there's some custom processing of project references to include them in the project itself: https://github.com/microsoft/fluentui/blob/master/packages/eslint-plugin/src/utils/configHelpers.js#L169. It appears perhaps that isn't being run for the vscode dev enviroment? Do you know how to have that run in the dev environment? Or alternatively, is it possible that my change here won't have a tangible impact on performance, since that code already appears to be adding all of the references to the project config?

Reply:
Ping @Hotell for the above comment 👆

For what it's worth, I tried making a PR with a no-op change (#27267) to see how the build times compare with this one.. and the total time was _less_ for this PR ("Fluent UI React - PR and CI" took 45m for this one; vs 54m for #27267). I don't think this PR actually makes the CI build faster, but it doesn't appear to make it _slower_.